src/formio/components/FileField.js

```diff
@@ -184,12 +185,8 @@ class FileField extends Formio.Components.components.file {
 
   upload(files) {
     if (this.component.multiple && this.component.maxNumberOfFiles) {
-      // this.data.file contains files that may have already been uploaded, while the argument 'files' gives the
-      // files that are being uploaded in this call
-      if (
-        files.length > this.component.maxNumberOfFiles ||
-        this.data?.file?.length >= this.component.maxNumberOfFiles
-      ) {
+      const existingFiles = get(this.data, this.component.key);
```

Comment:
There might be a formio util function to do this without using the lodash implementation detail, but couldn't find one

Reply:
`this.dataValue` is a getter that looks up the value using lodash in the base `Component` class, so I'd use that.